project/src/components/map/map.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { Icon, Marker } from 'leaflet';
+import {useRef, useEffect} from 'react';
+import 'leaflet/dist/leaflet.css';
+import { City, Offer } from '../../types/offers';
+import useMap from '../../hooks/use-map/use-map';
+
+type MapProps = {
+    city: City;
+    points: Offer[];
+    selectedPoint?: number | null;
+    className: string;
+  };
+
+const defaultCustomIcon = new Icon({
+  iconUrl: './img/pin.svg',
+  iconSize: [29, 40],
+  iconAnchor: [15, 40]
+});
+
+const currentCustomIcon = new Icon({
+  iconUrl: './img/pin-active.svg',
+  iconSize: [29, 40],
+  iconAnchor: [15, 40]
+});
+
+export default function Map({ className, city, points, selectedPoint }: MapProps): JSX.Element {
+  const mapRef = useRef(null);
```

Comment:
```suggestion
  const mapRef = useRef<HTMLElement | null>(null);
```
Странно, а тут TS не ругается на типы?
Из `null` невозможно понять какие типы будут использоваться, поэтому нужно указать их явно.